ui/app/src/Main.elm

```diff
@@ -25,21 +25,27 @@ import Views.AlertList.Types exposing (initAlertList)
 import Views.SilenceList.Types exposing (initSilenceList)
 import Views.SilenceView.Types exposing (initSilenceView)
 import Updates exposing (update)
+import Utils.Api as Api
+import Utils.Types exposing (ApiData(Loading))
+import Json.Decode as Json
 
 
-main : Program Never Model Msg
+main : Program Json.Value Model Msg
 main =
-    Navigation.program urlUpdate
+    Navigation.programWithFlags urlUpdate
         { init = init
         , update = update
         , view = Views.view
         , subscriptions = always Sub.none
         }
 
 
-init : Navigation.Location -> ( Model, Cmd Msg )
-init location =
+init : Json.Value -> Navigation.Location -> ( Model, Cmd Msg )
+init flags location =
     let
+        baseUrl =
+            location.pathname
```

Comment:
Can we change this to `basePath` or something similar? I think according to the [Wiki spec](https://en.wikipedia.org/wiki/URL#Syntax) the *url* is the entire string containing the *host* string as well (e.g. `github.com`), but we only want to get the *path* here (e.g. `prometheus/alertmanager`), right?